library/std/src/macros.rs

```diff
@@ -379,3 +379,51 @@ macro_rules! dbg {
         ($($crate::dbg!($val)),+,)
     };
 }
+
+/// Creates a [`HashMap`] containing the arguments.
+///
+/// This macro creates or either an empty [`HashMap`] using
+/// [`HashMap::new`] if there are no arguments, or generates
+/// a [`HashMap::insert`] call for each element.
+///
+/// This macro allows trailing commas.
+///
+/// This macro is used by concatenating comma separated sequences
+/// of `<key:expr> => <value:expr>`.
+///
+/// An example usage of this macro could be the following
+///
+/// ```rust
+/// #![feature(hash_map_macro)]
+///
+/// let map = hash_map!{
+///     "key" => "value",
+///     "key1" => "value1"
+/// };
+///
+/// for (key, value) in map {
+///     println!("{key:?} => {value:?}");
+/// }
+/// ```
+///
+/// Note that since this macro only generates [`HashMap::insert`]
+/// calls, the move semantics for the values are the same as
+/// the insert method.
```

Comment:
I'm also not sure whether this sentence adds useful information.
Maybe you can add a sentence like this at the end of the top paragraph:

> The key and value are moved into the hashmap.

although I would consider the ownership semantics here fairly obvious, but a short sentence like the one above probably wouldn't hurt.